### PR TITLE
Improve a few aspects of the install script

### PR DESCRIPTION
**Pull request by lbt**
There are a few commits which I thought would make the script more useful.

In particular I wanted to be able to hack on the code so installing 'editable' from a local clone rather than using pip install directly from git. This also allows the local repos to be re-checked-out to specific versions to solve pip dependency issues.

### CODE REVIEW

**Comment by Bulwagga** [comment]
These are nice ideas! One of my own goals for the script is that it will allow people to install OVOS onto a working Raspberry Pi image without needing to rebuild any existing environment.  I have a few dedicated systems that act as servers and data monitors and I'd like to give them voice control.  Anything in your changes that would cause problems for such systems?

**Comment by lbt** [comment]
No. That's my goal too. Things like using the venv and a dedicated user will support good isolation.

My next step is to create a remote dinkum-listener which will run on a few machines around the house that are already doing other things :)

**Comment by Bulwagga** [comment]
Got it.  Vision was also to keep things as simple as possible for development testing and getting new users up to speed quickly. Things like creating the python groups would make it easier for new users to understand the dependencies required when choosing among several listeners, for example. Builderjer can speak to the development environment considerations, I am coming from the user side.

**Review by builderjer — COMMENTED** [review]
I realize that the README could use some updating, but I don't think it should include the satellite option like that.  Maybe put it in the `manual_user_install.sh` script as an option?

File: manual_user_install.sh [hunk 1]
```diff
@@ -1,73 +1,76 @@
...
+    # padatious required to support newest ovos-core
+    # fann version must be fixed as raspbian libfann-dev is too old for latest
+    pip3 install padatious fann2==1.0.7
 
-    #Precise-lite wake-word (ww) cluster
-    pip3 install git+https://github.com/OpenVoiceOS/ovos-ww-plugin-precise
-    pip3 install git+https://github.com/OpenVoiceOS/ovos-ww-plugin-precise-lite
-    pip3 install --upgrade setuptools
+    # Precise-lite wake-word (ww) cluster
+    pip3 install --upgrade setuptools wheel pip
```
Comment (builderjer) [inline manual_user_install.sh]: Should this be moved uphill? As in the first thing installed and upgraded?

File: manual_user_install.sh [hunk 2]
```diff
@@ -146,30 +153,20 @@ function install_extra_skills (){
...
+    # install the OVOS extra skills
+    for p in $OVOS_EXTRA_SKILLS_REPOS; do
+	INSTALL="$INSTALL ${OVOS_SOURCE}/$p"
+    done
 
-    # OCP
-    pip3 install git+https://github.com/OpenVoiceOS/skill-ovos-news
-    pip3 install git+https://github.com/OpenVoiceOS/skill-ovos-somafm
-    pip3 install git+https://github.com/OpenVoiceOS/skill-ovos-youtube-music
+    # Add the NeonGeckoCom skills
```
Comment (builderjer) [inline manual_user_install.sh]: I think I want to remove the Neon skills from the image.  Most are in OVOS repos, and the ones that aren't don't play well with OVOS all of the time.  And they bring in extra dependencies such as `neon-utils`

File: manual_user_install.sh [hunk 3]
```diff
@@ -189,84 +186,141 @@ echo
...
-    echo "3. You can find pre-built OVOS/PI images at https://ovosimages.ziggyai.online/raspbian/"
-    echo ""
-    echo "4. After a reboot $HOME/.local/bin will be added to your path and give you access to the ovos command line utilities."
-    echo ""
-    echo "Enjoy your OVOS device"
 fi
+echo ""
+echo "1. Consider creating an .asoundrc and check your microphone with alsamixer, arecord, and aplay."
+echo ""
+echo "2. You can find documentation at https://openvoiceos.github.io/community-docs/install_raspbian/"
```
Comment (builderjer) [inline manual_user_install.sh]: This link is dead.  Maybe just point to the main page of the docs?

File: manual_user_install.sh [hunk 4]
```diff
@@ -189,84 +186,141 @@ echo
...
+echo ""
+echo "3. You can find pre-built OVOS/PI images at https://ovosimages.ziggyai.online/raspbian/"
```
Comment (builderjer) [inline manual_user_install.sh]: This is a wrong link also
should be `https://ovosimages.ziggyai.online/raspbian/development`
The way you have it shows several other image options also, which I guess is alright

File: manual_user_install.sh (hunk 3 above)
Comment (lbt) [reply manual_user_install.sh]: yes, I didn't write this - just changed the whitespace :)

File: manual_user_install.sh (hunk 4 above)
Comment (lbt) [reply manual_user_install.sh]: same - whitespace change :)

File: manual_user_install.sh (hunk 2 above)
Comment (lbt) [reply manual_user_install.sh]: OK - I didn't want to make any functional changes at this point

File: manual_user_install.sh (hunk 1 above)
Comment (lbt) [reply manual_user_install.sh]: yes - I just kept things in pretty much the same order initially

File: manual_user_install.sh (hunk 2 above)
Comment (builderjer) [reply manual_user_install.sh]: In fact, I did remove them from the image.  They are just still in the `install` script

**Comment by Bulwagga** [comment]
This has fallen behind during some exploratory work on admin-phal.

**Comment by builderjer** [comment]
I just changed the admis-phal service again. This is the only way I can get it to work correctly

File: manual_user_install.sh (hunk 2 above)
Comment (Bulwagga) [reply manual_user_install.sh]: I leave the Neon skills out of my builds... On a 3B+ I want to be as lightweight as possible. I didn't realize they were pulled from the image.

**Comment by Bulwagga** [comment]
Agree on the new admin-phal. This was the only substantial change in my most recent pull request (other than the links, etc.). I will cancel my PR to keep things simpler as you pull in jtb's work.

**Comment by builderjer** [comment]
If this works for you, @Bulwagga , I will merge with the suggested changes to the Readme especially

**Comment by Bulwagga** [comment]
Absolutely.  These seem like useful enhancements to me. In the meantime I'm giving thought to the ways raspovos is going to be used, and what features would be most useful for each.  Clearly there are developer users and we now have two of us that are interested in using OVOS an an overlay to support existing systems.  I can see a third cluster of users that simply want a smart device that protects their information.I'm thinking we want avoid building a "spork".  All the spearing capacity of the spoon  combined with the soup slurping capacity of the fork.From the outside looking in, I can see OVOS deeply embedded into the smart device if only to be lean and mean. At the same time I can see the overlay as a compartmentalized installation with something like venv. A dev environment would need maximum flexibility, not sure how to architect that.Then prioritize development tasks from those product descriptions.I'm excited to see these changes going in!+}
-------- Original message --------From: builderjer ***@***.***> Date: 8/24/23  7:25 PM  (GMT-05:00) To: OpenVoiceOS/raspbian-ovos ***@***.***> Cc: Bulwagga ***@***.***>, Mention ***@***.***> Subject: Re: [OpenVoiceOS/raspbian-ovos] WIP - improve a few aspects of the
  install script (PR #76) 
If this works for you, @Bulwagga , I will merge with the suggested changes to the Readme especially

**Comment by builderjer** [comment]
I am going to merge this after the suggested changes, and this discussion needs to be addressed more [here](https://github.com/orgs/OpenVoiceOS/discussions/185)

File: README.md [hunk 5]
```diff
@@ -14,6 +14,27 @@ This guide was originally designed for a headless, (No GUI) Raspberry Pi 3. The
 
 Please, if any mistakes, including spelling mistakes are found, or if a more detailed explanation of a step is needed, open an issue here and I will address it ASAP.
 
+## Usage
+
+Run `./manual_user_install.sh` for a full install which will work locally
+
+To just install a partial set of the modules set the unneeded ones to a space in the environment. This example will install only the modules for a satellite listener:
```
Comment (builderjer) [inline README.md]: This section should be removed from the README.  Connecting this way, while works, is strongly discouraged.

**Comment by lbt** [comment]
Let me know if I missed anything @builderjer

**Comment by builderjer** [comment]
Looks good!!

Thanks for the help

**Pull request merged**